db/src/commonMain/kotlin/com/islamversity/db/datasource/AyaLocalDataSource.kt

```diff
@@ -0,0 +1,115 @@
+package com.islamversity.db.datasource
+
+import com.islamversity.db.*
+import com.islamversity.db.model.*
+import com.islamversity.db.model.Aya
+import com.islamversity.db.model.Calligraphy
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.CoroutineContext
+
+interface AyaLocalDataSource {
+    suspend fun insertAya(
+        aya: AyaWithFullContent,
+        ayaCalligraphy: No_index_calligraphy,
+        context: CoroutineContext = Dispatchers.Default
+    )
+
+    suspend fun insertAyas(
+        ayas: List<AyaWithFullContent>,
+        ayaCalligraphy: No_index_calligraphy,
+        context: CoroutineContext = Dispatchers.Default
+    )
+
+    fun observeAllAyasForSora(
+        soraId: EntityUUID,
+        calligraphy: Calligraphy,
+        context: CoroutineContext = Dispatchers.Default
+    ): Flow<List<Aya>>
+
+    fun observeAllAyasForSora(
+        soraOrder: Long,
+        calligraphy: Calligraphy,
+        context: CoroutineContext = Dispatchers.Default
+    ): Flow<List<Aya>>
+
+    fun getAyaWithId(
+        entityId: EntityUUID,
+        calligraphy: Calligraphy,
+        context: CoroutineContext = Dispatchers.Default
+    ): Flow<Aya?>
+}
+
+class AyaLocalDataSourceImpl(
```

Comment:
Dagger has not been used!

Reply:
can't use it.
remember in kotlin.common package we can not reference any platform specifics. and `@Inject` is Java only.
We will provide all of the classes that are in `common` package using Dagger modules.